arangod/RocksDBEngine/RocksDBDumpContext.cpp

```diff
@@ -312,6 +367,9 @@ std::shared_ptr<RocksDBDumpContext::Batch const> RocksDBDumpContext::next(
     return it->second;
   }
 
+  // release lock temporarily because _channel.pop() can block
+  guard.unlock();
+
```

Comment:
Releasing this lock can cause a retried request to associate another batch with a already used batchId.

For example, assume on thread blocked on a call `next(5, 4)` on `_channel.pop`. Now, because of some network issues, arangodump retries the same request. We get another thread that enters `next(5, 4)`.

Now we have two threads, each will pop a new batch (say `A` and `B`) from the channel, but both associate this batch with id `5`. Assume that `B` is associated with `5` and the response for `A` is lost. Then first we get an error message that we have a duplicate batch id (although everything went fine) and batch `A` can never be received.

Reply:
Pushed a change that removes the unlock.